src/main/antlr/LibSLParser.g4

```diff
@@ -100,19 +102,60 @@ enumSemanticType
    ;
 
 enumSemanticTypeEntry
-   :    Identifier COLON expressionAtomic SEMICOLON
+   :   Identifier COLON expressionAtomic SEMICOLON
+   ;
+
+/* annotation declaration
+ * syntax: annotation Something(
+ *             variable1: int = 0,
+ *             variable2: int = 1
+ *         );
+ */
+annotationDecl
+   :   ANNOTATION name=Identifier (L_BRACKET (annotationDeclParams) (COMMA)? R_BRACKET)? SEMICOLON
```

Comment:
I'd extract a part with `L_BRACKET (annotationDeclParams) (COMMA)?` to a separate rule